firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -329,4 +332,17 @@ private String getFingerprintHashForPackage() {
       return null;
     }
   }
+
+  /**
+   * Returns parsed token expiration timestamp in seconds.
+   *
+   * @param expiresIn is expiration timestamp in String format: 604800s
+   */
+  private static long parseTokenExpirationTimestamp(String expiresIn) {
+    checkArgument(
+        EXPIRATION_TIMESTAMP_PATTERN.matcher(expiresIn).find(), "Invalid Expiration Timestamp.");
```

Comment:
Does find mean finding a sub string?

I guess you expect a exact match? not just find?